### PR TITLE
MM-14873 Add API command to create and delete a team and channel

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add API commands to create and delete a team and channel.

This fixes the issue at `channel_switcher_spec` where it sometimes failed when a hardcoded channel is not present in the initial sample data.  So such spec is done by creating a channel on pre-setup.

This is submitted on top of https://github.com/mattermost/mattermost-webapp/pull/2600.  Particular commit of this PR is at https://github.com/mattermost/mattermost-webapp/commit/633373e91197d096cf7bb8e663377c47142f1e7e.

#### Ticket Link
Jira ticket: [MM-14873](https://mattermost.atlassian.net/browse/MM-14873)

### CODE REVIEW

**Comment by saturninoabril** [comment]
Yes, it is.  I also mentioned in the ticket, will rebase once the other gets merged.

File: cypress/support/api_commands.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        method: 'POST',
+        body: {
+            team_id: teamId,
+            name: uniqueName,
+            display_name: displayName,
+            type,
+            purpose,
+            header,
+        },
+    }).then((response) => {
```
Comment (cometkim) [inline cypress/support/api_commands.js]: Is it necessary to explicitly yield with `then(res => res)`?

Comment (saturninoabril) [reply cypress/support/api_commands.js]: Oh, nice catch!  I initially added that to log what's being returned from that API.  Forgot to remove it, thanks!
Please take a look again.

File: cypress/support/api_commands.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+// https://api.mattermost.com/#tag/preferences
+// *****************************************************************************
+
+/**
+ * Saves user's preference directly via API
+ * This API assume that the user is logged in and has cookie to access
+ * @param {Array} preference - a list of user's preferences
+ */
+Cypress.Commands.add('apiSaveUserPreference', (preferences = []) => {
+    cy.request({
```
Comment (cometkim) [inline cypress/support/api_commands.js]:
```suggestion
    return cy.request({
```

File: cypress/support/api_commands.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    });
+});
+
+/**
+ * Saves channel display mode preference of a user directly via API
+ * This API assume that the user is logged in and has cookie to access
+ * @param {String} value - Either "full" (default) or "centered"
+ */
+Cypress.Commands.add('apiSaveChannelDisplayModePreference', (value = 'full') => {
+    cy.getCookie('MMUSERID').then((cookie) => {
```
Comment (cometkim) [inline cypress/support/api_commands.js]:
```suggestion
    return cy.getCookie('MMUSERID').then((cookie) => {
```

File: cypress/support/api_commands.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+        const preference = {
+            user_id: cookie.value,
+            category: 'display_settings',
+            name: 'channel_display_mode',
+            value,
+        };
+
+        cy.apiSaveUserPreference([preference]);
```
Comment (cometkim) [inline cypress/support/api_commands.js]:
```suggestion
        return cy.apiSaveUserPreference([preference]);
```

File: cypress/support/api_commands.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+        method: 'POST',
+        body: {login_id: user.username, password: user.password},
+    });
+});
+
+/**
+ * Logout a user directly via API
+ */
+Cypress.Commands.add('apiLogout', () => {
+    cy.request({
```
Comment (cometkim) [inline cypress/support/api_commands.js]:
```suggestion
    return cy.request({
```

**Comment by saturninoabril** [comment]
We don't have test that requires for it at the moment but we'll definitely have in the future.  So I update those per your comments, thanks @cometkim.

**Comment by jasonblais** [comment]
@saturninoabril Confirming this change is only for Cypress end-to-end testing, not anything a developer/user/admin would use directly with Mattermost?

**Comment by saturninoabril** [comment]
@jasonblais Yes, this is only for Cypress end-to-end testing.

**Pull request merged**